src/index.js

```diff
@@ -0,0 +1,144 @@
+  const weather = skySelect.value;
+
+  const skyClassMap = {
+    Sunny: 'sky-sunny',
+    Cloudy: 'sky-cloudy',
+    Rainy: 'sky-rainy',
+    Snowy: 'sky-snowy',
+    Windy: 'sky-windy',
+    Foggy: 'sky-foggy'
+  };
+
+  const newClass = skyClassMap[weather];
+  const allBoxes = document.querySelectorAll('.box');
+
+  allBoxes.forEach((box) => {
+    box.classList.remove(
+      'sky-sunny',
+      'sky-cloudy',
+      'sky-rainy',
+      'sky-snowy',
+      'sky-windy',
+      'sky-foggy'
+    );
+    box.classList.add(newClass);
+  });
+};
+
+const updateWeatherEmoji = () => {
+  const skySelect = document.getElementById('skySelect');
+  const weatherEmoji = document.getElementById('weatherEmoji');
+  const weather = skySelect.value;
+
+  const emojiMap = {
+    Sunny: '☀️🌈☀️☀️🌈☀️☀️🌈☀️☀️🌈☀️☀️',
+    Cloudy: '☁️⛅☁️☁️⛅☁️☁️⛅☁️☁️⛅☁️☁️',
+    Rainy: '🌧️🌦️🌧️🌧️🌦️🌧️🌧️🌦️🌧️🌧️🌦️🌧️🌧️',
+    Snowy: '❄️🌨️❄️❄️🌨️❄️❄️🌨️❄️❄️🌨️❄️❄️',
+    Windy: '🌬️🍃🌬️🌬️🍃🌬️🌬️🍃🌬️🌬️🍃🌬️🌬️',
+    Foggy: '🌫️🌁🌫️🌫️🌁🌫️🌫️🌁🌫️🌫️🌁🌫️🌫️'
+  };
+
+  weatherEmoji.textContent = emojiMap[weather];
+};
+
+const updateCityName = () => {
+  const input = document.getElementById('city-input');
+  const display = document.getElementById('displayed-city-name');
+  display.textContent = input.value || "Seattle";
+};
+
+const resetCityName = () => {
+  const input = document.getElementById('city-input');
+  input.value = "Seattle";
+  updateCityName();
+};
+
+const registerEventHandlers = () => {
+  document.getElementById('increaseTempControl').addEventListener('click', increaseTemp);
+  document.getElementById('decreaseTempControl').addEventListener('click', decreaseTemp);
+  document.getElementById('skySelect').addEventListener('change', () => {
+    updateWeatherBackground();
+    updateWeatherEmoji();
+  });
+  document.getElementById("city-input").addEventListener('input', updateCityName);
+  document.getElementById("choose-place").addEventListener('click', resetCityName);
+
+  document.getElementById('temp-now-btn').addEventListener('click', () => {
+    const cityNameInput = document.getElementById('city-input');
+    const city = cityNameInput.value || 'Seattle';
+
+    axios.get('http://127.0.0.1:5000/location', {
+      params: { q: city }
+    })
+      .then((response) => {
+        const { lat, lon } = response.data[0];
+        return axios.get('http://127.0.0.1:5000/weather', {
+          params: { lat, lon }
+        });
+      })
+      .then((response_weather) => {
+        const kelvinTemperature = response_weather.data.main.temp;
+        state.tempValue = Math.round((kelvinTemperature - 273.15) * 9/5 + 32);
+        updateTempValue();
+      })
+      .catch((error) => {
+        console.error("Error fetching weather data:", error);
+      });
+  });
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+  registerEventHandlers();
+  updateTempValue();
+  updateWeatherBackground();
+  updateWeatherEmoji();
+  updateCityName();
+});
```

Comment:
I'd prefer you create a `onLoaded` helper function because the descriptive name tells others that the function will run on page load.

```suggestion
const onLoaded = () => {
registerEventHandlers();
  updateTempValue();
  updateWeatherBackground();
  updateWeatherEmoji();
  updateCityName();
};

document.addEventListener('DOMContentLoaded', onLoaded);
```